### ISSUE TITLE
[camelcase] false positive on interface properties

### ISSUE THREAD

**Issue by bradzacher** [bradzacher/eslint-plugin-typescript#177]
**Repro**

```JSON
{
  "rules": {
    "camelcase": ["error", { "properties": "never" }]
  }
}
```

```TS
interface Foo { b_ar: null }
```

**Expected Result**
No errors

**Actual Result**
`Identifier 'b_ar' is not in camel case.`

**Additional Info**
Moved from unresponsive PR #95
https://github.com/eslint/typescript-eslint-parser/issues/492

**Comment by bradzacher** [comment]
this is a huge problem (i'm surprised more people don't report it).

I dug into this a while ago, and only just remember where I put the issue (I've updated the OP with the link)

**Comment by weirdpattern** [comment]
Just read the bug you linked. I think extending the base ESLint rule to handle typescript specific nodes makes the most sense as we don't want to impact other rules that use the Identifier node. 

We have done it before #68 

Thoughts?

**Comment by bradzacher** [comment]
sounds good to me!

I don't remember if you can do this easily via eslint's context or not, but if you can, it should be as easy as interrogating the parent of the identifier!

Also considering we only currently have a minor modification - I wonder if it's possible to just import the existing eslint rule and do something like:

```
if (isTsInterfaceProp() && settings.properties === 'always') {
   eslintCamelcase()
}
```
